### ISSUE TITLE
Sftp LIBSSH2_ERROR_* caused panic during Drop

### ISSUE THREAD

**Issue by crisidev** [alexcrichton/ssh2-rs#180]
Hello,
I am facing an issue with Sftp subsystem where if any protocol error happens during Drop, it causes a panic which is not catchable, resulting in undefined errors.

Example:
```rust
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `-7`,
 right: `0`', /home/bigo/.cargo/.../ssh2-0.8.0/src/sftp.rs:645:13
```

This specific example is trying to return `LIBSSH2_ERROR_SOCKET_SEND`

Is there any reason why this piece of code does not return an Error instead of just panicing?

Further more, is there any way for me to catch this panic? I have tried unwinding the stack, but I can't make it work..

The panic can come from these 2 assertions inside SFTP: 
* https://github.com/alexcrichton/ssh2-rs/blob/master/src/sftp.rs#L435
* https://github.com/alexcrichton/ssh2-rs/blob/master/src/sftp.rs#L645

**Comment by wez** [comment]
> Is there any reason why this piece of code does not return an Error instead of just panicing?

We can't change the signature of the `Drop` trait :-/

@spebern: what do you think about removing the asserts from these locations?  I don't think there is any sane error recovery that can be done, so it is probably fine to ignore this error.

**Comment by crisidev** [comment]
Silly me by asking about returning an error.. you cannot change this trait signature, of course.. 

Regarding the safety of ignoring this error, I think it is fine.. obviously something went catastrophically wrong, but we are very late in the execution here and there is nothing you can really do.

It would be nice to have some sort of notification thou that something went really wrong..

**Comment by spebern** [comment]
@wez I don't think those asserts should be necessary

**Comment by crisidev** [comment]
Allright, I'll work on a PR today.

**Comment by wez** [comment]
This was published as part of ssh2 `0.8.1`

**Comment by crisidev** [comment]
Thanks a lot @wez!
